clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -506,6 +519,22 @@ abstract class OpenstackClientProvider {
   }
 
   /**
+   * Get all instance ids of server resources associated with a stack.
+   * @param region
+   * @param stackName
+   */
+  List<String> getInstanceIdsForStack(String region, String stackName) {
+    List<String> ids = []
+    List<? extends Resource> resources = handleRequest {
+      getRegionClient(region).heat().resources().list(stackName)
+    }
+    for (Resource r : resources) {
```

Comment:
Can use: `List<String> ids = resources.findResults { ... }` instead of the for-each loop.